### PR TITLE
Gui work

### PR DESCRIPTION
**Pull request by petricafinaga**

### CODE REVIEW

File: src/chat/client/ClientAgent.java [hunk 1]
```diff
@@ -42,6 +42,7 @@ protected void setup() {
 			Utils.WriteClientConfigToFile(clientConfig, configFileName);
 		}
 
+
```
Comment (petricafinaga) [inline src/chat/client/ClientAgent.java]: remove this line

File: src/chat/client/ClientAgent.java [hunk 2]
```diff
@@ -64,14 +65,20 @@ protected void takeDown() {
 	/* End override methods from inherited class */
 
 	public void OnAllClients(ChatClient[] clients) {
+		// TO DO
+
```
Comment (petricafinaga) [inline src/chat/client/ClientAgent.java]: remove

File: src/chat/client/ClientAgent.java [hunk 3]
```diff
@@ -64,14 +65,20 @@ protected void takeDown() {
...
 
 	public void OnAllClients(ChatClient[] clients) {
+		// TO DO
+
 		clientGui.GUIAddUsers(clients);
 	}
 
 	public void OnClientUpdate(ChatClient client) {
+		// TO DO
+
```
Comment (petricafinaga) [inline src/chat/client/ClientAgent.java]: remove

File: src/chat/client/ClientAgent.java [hunk 4]
```diff
@@ -64,14 +65,20 @@ protected void takeDown() {
...
 		clientGui.GUIAddOrModifyUserStatus(client);
 	}
 
 	public void OnTextMessage(String clientName, String messageText) {
+		// TO DO
+
```
Comment (petricafinaga) [inline src/chat/client/ClientAgent.java]: remove

File: src/chat/client/ClientAgent.java [hunk 5]
```diff
@@ -102,6 +109,7 @@ private void SubscribeToServer() {
 		final ACLMessage message = new ACLMessage(ACLMessage.INFORM);
 
 		message.addReceiver(serverAid);
+
```
Comment (petricafinaga) [inline src/chat/client/ClientAgent.java]: remove

File: src/chat/client/ClientGUI.java [hunk 6]
```diff
@@ -7,35 +7,40 @@
...
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.border.EmptyBorder;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+import javax.swing.JTextPane;
+import javax.swing.ListSelectionModel;
+import javax.swing.JButton;
+
```
Comment (petricafinaga) [inline src/chat/client/ClientGUI.java]: Please put imports in alphabetical order, as they were before.

File: src/chat/client/ClientGUI.java [hunk 7]
```diff
@@ -67,34 +71,29 @@
...
-//					frame.setVisible(true);
-//				} catch (Exception e) {
-//					e.printStackTrace();
-//				}
-//			}
-//		});
-//	}
+	Map<String, DefaultStyledDocument> usersMessages;
+
+	String clientAlias;
```
Comment (petricafinaga) [inline src/chat/client/ClientGUI.java]: access modifiers??

File: src/chat/client/ClientGUI.java [hunk 8]
```diff
@@ -67,34 +71,29 @@
...
-			// a.UpdateAlias(alias);
+			clientAlias = JOptionPane.showInputDialog(contentPane, "Enter your username");
+			a.UpdateAlias(clientAlias);
+		} else {
+			clientAlias = alias;
 		}
 
+		usersMessages = new HashMap<String, DefaultStyledDocument>();
+
+		this.setTitle(clientAlias);
```
Comment (petricafinaga) [inline src/chat/client/ClientGUI.java:89]: maybe `setResizable(false);` can be here, above this line??

File: src/chat/client/ClientGUI.java [hunk 9]
```diff
@@ -245,6 +244,13 @@ public void GUIAddUsers(ChatClient[] clients) {
 			data.add(chatClient.getStatus().toString());
 			data.add(chatClient.getName());
 			usersTableModel.addRow(data);
+//			Add messages into the list
+			try {
+				usersMessages.put(chatClient.getAlias(), new DefaultStyledDocument());
+			} catch (Exception e) {
+				// TODO: handle exception
```
Comment (petricafinaga) [inline src/chat/client/ClientGUI.java]: da pula

File: src/chat/client/ClientGUI.java [hunk 10]
```diff
@@ -263,7 +269,14 @@ public void GUIAddOrModifyUserStatus(ChatClient client) {
 			data.add(client.getStatus().toString());
 			data.add(client.getName());
 			usersTableModel.addRow(data);
+			try {
+				usersMessages.put(client.getAlias(), new DefaultStyledDocument());
+			} catch (Exception e) {
+				// TODO: handle exception
```
Comment (petricafinaga) [inline src/chat/client/ClientGUI.java]: pula here too

File: src/chat/client/ClientGUI.java [hunk 11]
```diff
@@ -263,7 +269,14 @@ public void GUIAddOrModifyUserStatus(ChatClient client) {
...
 			data.add(client.getName());
 			usersTableModel.addRow(data);
+			try {
+				usersMessages.put(client.getAlias(), new DefaultStyledDocument());
+			} catch (Exception e) {
+				// TODO: handle exception
+				e.printStackTrace();
+			}
 		}
+		updatesLabel.setText("");
```
Comment (petricafinaga) [inline src/chat/client/ClientGUI.java]: dc mai pui si asta daca dupa setezi textu din nou?

File: src/chat/client/ClientAgent.java (hunk 5 above)
Comment (tinkode56) [reply src/chat/client/ClientAgent.java]: why remove this?

File: src/chat/client/ClientAgent.java [hunk 12]
```diff
@@ -39,9 +39,9 @@ protected void setup() {
 
 		if (clientConfig == null) {
 			clientConfig = ClientConfig.GetDefaultClientConfig();
-			Utils.WriteClientConfigToFile(clientConfig, configFileName);
```
Comment (petricafinaga) [inline src/chat/client/ClientAgent.java]: Please add this line again

**Pull request merged**